src/common/utils.hpp

```diff
@@ -687,6 +689,31 @@ int getenv_int_user(const char *name, int default_value = 0);
 // "DNNL_" (secondary) prefixes.
 std::string getenv_string_user(const char *name);
 
+// These are locale-invariant wrappers to define streaming objects for
+// string manipulation. Use these instead of the std library variants, namely,
+// std::stringstream, std::istringstream and std::ostringstream to ensure
+// locale-independent string behavior.
+struct dnnl_stringstream_t : public std::stringstream {
+    dnnl_stringstream_t() { this->imbue(std::locale::classic()); }
+    dnnl_stringstream_t(const std::string &str) : std::stringstream(str) {
+        this->imbue(std::locale::classic());
+    }
+};
```

Comment:
It'd be nice to keep the interfaces as close to the originals as possible while avoiding any further changes to the locale.

```suggestion
struct dnnl_stringstream_t : public std::stringstream {
    template <typename ... Args>
    dnnl_stringstream_t(Args &&...args) : std::stringstream(std::forward<Args>(args)...) {
        imbue(std::locale::classic());
    }

private:
    using stringstream::imbue;
};
```